smoke_test/steps/base_step.rb

```diff
@@ -0,0 +1,41 @@
+module SmokeTest
+  module Steps
+    class BaseStep
+      include Capybara::DSL
+
+      attr_accessor :state
+
+      def initialize(state)
+        puts "Step:#{step_name}"
+        @state = state
+      end
+
+      def validate!
+        true
+      end
+
+      def complete_step
+        fail NotImplementedError
```

Comment:
Inheriting from this class requires this method to be overriden, as a result of this we end up implementing a couple of empty methods. A better approach is probably to just have the superclass have the empty method and don't require the override & then remove the empty methods in the relevant step implemtations.